lib/reek/spec/smell_matcher.rb

```diff
@@ -0,0 +1,57 @@
+module Reek
+  module Spec
+    #
+    # Matches a +SmellWarning+ object agains a smell type and hash of attributes.
+    #
+    class SmellMatcher
+      attr_reader :smell_warning
+
+      COMPARABLE_ATTRIBUTES = %i(message lines context source)
+
+      def initialize(smell_warning)
+        @smell_warning = smell_warning
+      end
+
+      def matches?(klass, attributes = {})
+        matches_smell_type?(klass) && matches_attributes?(attributes)
+      end
+
+      def matches_smell_type?(klass)
+        smell_classes.include?(klass.to_s)
+      end
+
+      def matches_attributes?(attributes)
+        check_attributes_comparability(attributes)
+
+        main_attributes = attributes.slice(*COMPARABLE_ATTRIBUTES)
+        parameter_attributes = attributes.except(*COMPARABLE_ATTRIBUTES)
```

Comment:
Wat? I would _never_ suggest to infest a codebase with Active Support, [`Enumerable#partition` is core Ruby](http://www.rubydoc.info/stdlib/core/Enumerable#partition-instance_method).

I’m also not sure what you mean by ‘it returns an array of arrays’ – it does, that’s why I suggested a destructuring assign `comparables, params = …`.